### DISCUSSION TITLE
Upgrading from v4 to v5

### DISCUSSION THREAD

**Discussion by wit3** [spatie/laravel-permission#1851] [Q&A] [answered]
Hi,

The doc is empty for upgrading from v4 to v5, any advice or feedback?

[Upgrading.md](https://github.com/spatie/laravel-permission/blob/2512d5a22aaae0bab8d62a02e502a603776da9da/docs/upgrading.md)

**Comment by erikn69** [comment, marked as answer]
No longer flush cache on User role/perm assignment changes #1832 
Change default-guard-lookup to prefer current user's guard #1817
Teams/Groups feature #1804
Customized pivots instead of role_id,permission_id #1823

**Comment by gabidj** [comment]
You can check the changelog, it does point out all the changes: https://github.com/spatie/laravel-permission/blob/main/CHANGELOG.md
